libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/statistics/KotlinBuildStatsService.kt

```diff
@@ -106,13 +106,21 @@ internal abstract class KotlinBuildStatsService internal constructor() : BuildAd
                             )
                             instance = JMXKotlinBuildStatsService(mbs, beanName)
                         } else {
+                            val kotlinBuildStatProvider = project.provider{ KotlinBuildStatListener(beanName) }
                             val newInstance = DefaultKotlinBuildStatsService(gradle, beanName)
+
+                            if (isConfigurationCacheAvailable(gradle)) {
+                                listenerRegistryHolder.listenerRegistry!!.onTaskCompletion(kotlinBuildStatProvider)
```

Comment:
❌ This registration isn't valid and won't do anything. I would simply remove it for now.

See https://github.com/JetBrains/kotlin/pull/3521/files#r452728632